L1Trigger/TrackFindingTracklet/src/FitTrack.cc

```diff
@@ -1020,6 +1023,53 @@ void FitTrack::execute(unsigned int iSector) {
         trackfit_->addTrack(bestTracklet);
       }
     }
+    // store bit and clock accurate TB output
+    if (settings_.storeTrackBuilderOutput() && bestTracklet) {
+      // add gap if enough layer to form track
+      static constexpr int limit = 4;
+      if ((int)bestTracklet->getL1Stubs().size() < limit) {
+        streamTrack.emplace_back(tt::Frame());
+        for (auto& stream : streamsStub)
+          stream.emplace_back(tt::FrameStub());
+        continue;
+      }
+      // convert Track word
+      const int seedType = bestTracklet->getISeed();
+      static const string valid = "1";
+      const string seed = TTBV(seedType, settings_.nbitsseed()).str();
+      const string rinv = bestTracklet->fpgarinv().str();
+      const string phi0 = bestTracklet->fpgaphi0().str();
+      const string z0 = bestTracklet->fpgaz0().str();
+      const string t = bestTracklet->fpgat().str();
+      streamTrack.emplace_back(valid + seed + rinv + phi0 + z0 + t);
+      // hitMap used to remember whcih layer had no stub to fill them with gaps
+      TTBV hitMap(0, channelAssignment->maxNumProjectionLayers());
+      // convert and fill stubs on this track into streamsStub
+      for (const auto& stub : bestTracklet->getL1Stubs()) {
+        // get TTStubRef of this stub
+        const TTStubRef& ttStubRef = stub->ttStubRef();
+        // get layerId and skip over seeding layer
+        int layerId(-1);
+        if (!channelAssignment->layerId(seedType, ttStubRef, layerId))
+          continue;
+        // mark layerId
+        hitMap.set(layerId);
+        // tracklet layerId
+        const int trackletLayerId = channelAssignment->trackletLayerId(ttStubRef);
+        // get stub Residual
+        const Residual& resid = bestTracklet->resid(trackletLayerId);
+        // create bit accurate 64 bit word
+        const string& r = resid.stubptr()->r().str();
+        const string& phi = resid.fpgaphiresid().str();
+        const string& rz = resid.fpgarzresid().str();
+        // store TTStubRef and bit accurate 64 bit word in clock accurate output
+        streamsStub[layerId].emplace_back(ttStubRef, valid + r + phi + rz);
```

Comment:
The contents of these streams is similar to that written to txt file by TrackFitMemory::writeTF(). The main differences are:
1) The stream is created before the data is stored in the TrackFitMemory, rather than after. Is this because the memory suppresses the gaps?
2) Some elements of the agreed TrackBuilder & DR output https://twiki.cern.ch/twiki/bin/viewauth/CMS/HybridDataFormat#TrackWord_StubWord_written_by_Tr , such as the Stub ID have been dropped. If you think that these are not needed, we should propose changing the twiki, though the Stub ID is needed by the DR. (Are the comments incorrect, and this code is actually writing streams from the end of the tracklet chain, rather than after the TrackBuilder?).

Reply:
1. yes 2. those words are not needed if you connect TB with KF without PD in between. Without PD the TB output is the end of tracklet chain.